lib/geokit/geocoders/mapbox.rb

```diff
@@ -68,6 +67,9 @@ def self.set_address_components(result_json, loc)
         if result_json["place_name"]
           loc.full_address = result_json["place_name"]
         end
+        if !loc.city && result_json["id"] =~ /^city\./
+          loc.city = result_json['text']
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.